### ISSUE TITLE
Support the Servlet API

### ISSUE THREAD

**Issue by zgranik** [micronaut-projects/micronaut-core#1002]
We are trying to implement OData using SDL OData Framework but we have a problem with mapping **io.micronaut.http.HttpRequest** to **javax.servlet.http.HttpServletRequest** in order to handle 

```
@Log4j2
@Controller(value = "/odata.svc{/path:.*}")
public class ApiActivityLogController extends AbstractODataController {
```

Any suggestions on how we can make HttpServletRequest out of Micronaut HttpRequest?

**Comment by graemerocher** [comment]
Maybe you can elaborate a bit more, why do you need a HttpServletRequest?

**Comment by zgranik** [comment]
SDL OData framework expects HttpServletRequest to be forwarded but Micronaut has is own HttpRequest classes it is not using javax. For example:

`@Log4j2
@Controller(value = "/odata.svc{/path:.*}")
public class ApiActivityLogController extends AbstractODataController {

    @Get
    @Put
    @Post
    @Patch
    @Delete
    protected void service(HttpRequest<?> request)
            throws ServletException, IOException {
        log.info("Us'o!");
        log.info(request);

        super.service((HttpServletRequest) request, null);
    }

}`

**Comment by geemang2000** [comment]
Along with this how to get a HttpServletResponse. There are existing APIs that need this traditional Servlet Types.

**Comment by zgranik** [comment]
Yes, so we solved this problem by mapping NettyMutableHttpResponse to HttpServletResponse not pretty nice way but we did not find any other. Because there are a lot of frameworks etc that require javax.servlet.http.HttpServletResponse/Request.

@graemerocher I am wondering how safe is for us to use a Micronaut framework for re-architecting a large project and hope that you will continue to develop and improve Micronaut framework?
